### PR TITLE
Quirks misc code changes

### PR DESCRIPTION
**Pull request by SJuliez**
Looking at allowing Quirk editing in MML, so I really needed some API improvements for Quirks and WeaponQuirks. Also updates and condenses the Quirks validity checks.

In the MechSummary, the weapon quirks list now shows the weapon location. In the weapon list, weapons with quirks get a (Q) marker.

image

### CODE REVIEW

File: megamek/src/megamek/common/options/IOption.java [hunk 1]
```diff
@@ -71,66 +71,66 @@
...
-    public abstract String stringValue();
+    String stringValue();
 
     /**
      * Sets the value
      * 
      * @param value value to set
      */
-    public abstract void setValue(Object value);
+    void setValue(Object value);
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/common/options/IOption.java:101]: legacy issue. Seems no problem at this moment. I tried, can't just remove it.

**Comment by HammerGS** [comment]
I've been testing the Quirks branch's and everything seems to work. But I haven't tested a unit with every quirk.

File: megamek/src/megamek/common/options/IOption.java [hunk 2]
```diff
@@ -29,27 +29,27 @@ public interface IOption extends IBasicOption, IOptionInfo {
     /**
      * Boolean option type
      */
-    public static final int BOOLEAN = 0;
+    int BOOLEAN = 0;
```
Comment (NickAragua) [inline megamek/src/megamek/common/options/IOption.java:32]: Why make these non-static and modifiable?

File: megamek/src/megamek/common/options/Quirks.java [hunk 3]
```diff
@@ -160,530 +150,140 @@ public synchronized void initialize() {
...
-                    || qName.equals(OptionsConstants.QUIRK_POS_OVERHEAD_ARMS)   
-                    || qName.equals(OptionsConstants.QUIRK_POS_PRO_ACTUATOR)
-                    || qName.equals(OptionsConstants.QUIRK_POS_BATTLE_FIST_LA)
-                    || qName.equals(OptionsConstants.QUIRK_POS_BATTLE_FIST_RA)
-                    || qName.equals(OptionsConstants.QUIRK_POS_VESTIGIAL_HANDS_LA)
-                    || qName.equals(OptionsConstants.QUIRK_POS_VESTIGIAL_HANDS_RA)) {
-                return false;
-            }
-            return true;
+            return quirk.isAnyOf(QUIRK_NEG_HARD_PILOT, QUIRK_NEG_BAD_REP_IS, QUIRK_NEG_BAD_REP_CLAN,
```
Comment (NickAragua) [inline megamek/src/megamek/common/options/Quirks.java:232]: Not really sure this is any better for readability (from my perspective it's a little easier to have one quirk per line), but not really a deal breaker either way.

**Review by NickAragua — COMMENTED** [review]
One or two loose questions/thoughts.

File: megamek/src/megamek/common/options/IOption.java (hunk 2 above)
Comment (SJuliez) [reply megamek/src/megamek/common/options/IOption.java:32]: It's in an interface, so they're public static final automatically. IntelliJ warns about it and I tend to remove redundant clutter, but that's certainly subjective.
:grin:
image

File: megamek/src/megamek/common/options/Quirks.java (hunk 3 above)
Comment (SJuliez) [reply megamek/src/megamek/common/options/Quirks.java]: I agree but I also strongly dislike scrolling through hundreds of lines. This is better, isn't it?

image

Comment (HammerGS) [reply megamek/src/megamek/common/options/Quirks.java]: If the data guy can follow it, all is good :)

**Comment by SJuliez** [comment]
Hehe I gave myself some conflicts :)

**Review by NickAragua — APPROVED** [review]

File: megamek/src/megamek/common/options/Quirks.java (hunk 3 above)
Comment (NickAragua) [reply megamek/src/megamek/common/options/Quirks.java]: Fine by me.

File: megamek/src/megamek/common/options/IOption.java (hunk 2 above)
Comment (NickAragua) [reply megamek/src/megamek/common/options/IOption.java:32]: Good to know.

**Pull request merged**